client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -0,0 +1,110 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { debounce, find, get, noop, trim } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import SuggestionSearch from 'components/suggestion-search';
+import { getHttpData, requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
+
+export class SiteVerticalsSuggestionSearch extends Component {
+	static propTypes = {
+		initialValue: PropTypes.string,
+		onChange: PropTypes.func,
+		placeholder: PropTypes.string,
+	};
+
+	static defaultProps = {
+		initialValue: '',
+		onChange: noop,
+		placeholder: '',
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			searchValue: props.initialValue,
+			charsToTriggerSearch: 1,
+		};
+	}
+
+	onSiteTopicChange = value => {
+		value = trim( value );
+		const verticalData = find( this.props.verticals, { vertical_name: value } );
+		if (
+			value &&
+			value.length > this.state.charsToTriggerSearch &&
+			// Don't trigger a search if the input value is present in the verticals results.
+			! verticalData
+		) {
+			this.props.requestVerticals( value );
+		}
+		this.setState( { siteTopicValue: value } );
+		this.props.onChange(
+			verticalData || {
+				vertical_name: value,
+				vertical_slug: value,
+			}
+		);
+	};
+
+	getSuggestions = () => this.props.verticals.map( vertical => vertical.vertical_name );
+
+	render() {
+		const { translate, placeholder } = this.props;
+
+		return (
+			<SuggestionSearch
+				id="siteTopic"
+				placeholder={
+					placeholder || translate( 'e.g. Fashion, travel, design, plumber, electrician' )
```

Comment:
Looks like we don't have matching verticals for some of these suggested topics in the placeholder. There is no exact topic match for **design** and **travel**. 

I tested with Spanish that the matches are fewer. I got a match only for _moda_. We could have a similar problem in other Mag 16 languages.

<img width="559" alt="screenshot 2018-12-20 at 6 09 42 pm" src="https://user-images.githubusercontent.com/1269602/50285570-9c891580-0482-11e9-92e6-6cfa62f0664b.png">

Reply:
> Looks like we don't have matching verticals for some of these suggested topics in the placeholder

It's an important observation, but I think it's not a huge deal right now, especially given the fact that users can enter whatever they want.